pennylane/gradients/parameter_shift.py

```diff
@@ -23,6 +23,40 @@
 from .finite_difference import finite_diff, generate_shifted_tapes
 
 
+NONINVOLUTORY_OBS = {
+    "Hermitian": lambda obs: obs.__class__(obs.matrix @ obs.matrix, wires=obs.wires),
+    "SparseHamiltonian": lambda obs: obs.__class__(obs.matrix @ obs.matrix, wires=obs.wires),
+    "Projector": lambda obs: obs,
+}

```

Comment:
:100: It's great to have this place in case the list will further grow.

Reply:
Ideally, we won't have this in the future! Once the operator refactor is done, hopefully you should just be able to do `obs = obs @ obs` :)